src/main/java/net/glowstone/command/GameModeCommand.java

```diff
@@ -0,0 +1,108 @@
+package net.glowstone.command;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.GameMode;
+import org.bukkit.Location;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.defaults.BukkitCommand;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+import java.util.Collections;
+
+public class GameModeCommand extends BukkitCommand {
+    public GameModeCommand() {
+        super("gamemode", "Change the game mode of a player.", "/gamemode <mode> [player]", Collections.emptyList());
+        setPermission("minecraft.command.gamemode");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        if (!testPermission(sender)) return false;
+        if (args.length == 0 || args.length == 1 && !(sender instanceof Player)) {
+            sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+            return false;
+        }
+        String gm = args[0].toLowerCase();
+        GameMode gamemode = null;
+        switch (gm) {
```

Comment:
Although, in Vanilla, using an incorrect GameMode ID prints out `'x' is not a valid number!`.